lib/solveEquation/EquationOperations.js

```diff
@@ -21,13 +21,11 @@ const EquationOperations = {};
 // by swapping the right and left sides if it is only in the right side.
 // So 3 = x would become x = 3.
 EquationOperations.ensureSymbolInLeftNode = function(equation, symbolName) {
-  const leftSideSymbolTerm = Symbols.getLastSymbolTerm(
-    equation.leftNode, symbolName);
-  const rightSideSymbolTerm = Symbols.getLastSymbolTerm(
-    equation.rightNode, symbolName);
+  const hasSymbolOnLeft = containsSymbolNode(equation.leftNode, symbolName);
+  const hasSymbolOnRight = containsSymbolNode(equation.rightNode, symbolName);
```

Comment:
this block of code has been quite a headache at several points while using mathsteps, and I think my revisions here should put the wonkiness to rest... 

The purpose of the old lines was merely to check whether or not a particular side contained the relevant variable. To accomplish that, however, it ended up sending the nodes through the 'Polynomial' evaluator and a whole bunch of unnecessary stuff so that if it didn't meet the 'is a polynomial' check, it wouldn't return anything. 

All that this needs to do, however, is just return true or false if the given variable is present. 


By fixing this check, we actually fix the test case  ` ['g *( x ) = (x-4)^(2) - 3', 'g = x - 8 + 13 / x'],` so that's nice.